### PR TITLE
remove parser for using back-ticks for infix function application

### LINKED ISSUE #2229: Get rid of single backtick for infix function application
Discussed on Slack a while back but never created a ticket to track.

The pretty-printer doesn't have any way of knowing when to use it when rendering, so it seems a bit pointless, more like a shortcut. 

Also a `append` b can be rewritten `a |> append <| b`
```
append = (Text.++)

> "Howdy" |> append <| " partner!"
          |> append <| " Yee haw!!"
          |> append <| " Unison's got infix wordy already!"
          |> append <| " Do we really need special syntax?"
```
Once this is done, we could maybe start using single backtick within a doc in place of the double backtick example syntax.

This should be pretty easy - open up `TermParser.hs` and just delete the parser for this. And then update all the tests that are using the old syntax.

### PR DESCRIPTION
**Pull request by kolharsam**
Implements #2229.

### CODE REVIEW

**Comment by ChrisPenner** [comment]
Hi @kolharsam, thanks for getting us started on this one 😄 
Looks like there were a few backtick'd expressions being used in the internal source and in a few transcripts so I just went ahead and cleaned those up.

There was also some sneaky changes to make sure we can still parse infix operators like `Doc.++`, but I think it's ready to go now 👍🏼

**Comment by ChrisPenner** [comment]
@rlmark  I think there's a few spots in the language guide that'll need to be updated once this releases too;
starting from: https://www.unisonweb.org/docs/language-reference/#function-application

**Comment by kolharsam** [comment]
@ChrisPenner, I'm more than happy to help out on other issues as well! Please feel free to point me in some direction, or I could do some exploration myself. Thanks again!

File: unison-src/tests/trash2/reflog [hunk 1]
```diff
@@ -0,0 +1 @@
+sjg2v58vn2cb68ltf45hoqaj9eel8lb4hapougase6mbsbnfecvv2jl3bc500bat13vhec6kqf2tee74e6e9ahckukb6ch42bimr2pg hcfruc0iflihej5u2nih0ups834u64chbdh0eoci19t3i846j2s2a8u0936m01f5uv3cq51glr4unih0g3jrc75rtdt711872eqbi9o pull https://github.com/unisonweb/base:.releases._M2k .base
```
Comment (aryairani) [inline unison-src/tests/trash2/reflog]: Let's remove this file at least.

**Review by aryairani — DISMISSED** [review]
Seems okay to me once that trash file is deleted (@ChrisPenner) and once @kolharsam agrees to and adds himself to `CONTRIBUTORS.markdown`.

File: unison-src/tests/trash2/reflog (hunk 1 above)
Comment (aryairani) [reply unison-src/tests/trash2/reflog]: @kolharsam I guess could you remove this file, since you’ve got the branch checked out anyway?

**Review by aryairani — APPROVED** [review]

**Comment by aryairani** [comment]
@ceedubs @stew Any objections to merging this? i.e. will it break your all-in-one file to lose backticked infix function application?  If not, please go ahead and merge it. Thanks!

**Comment by ceedubs** [comment]
@aryairani I don't think that this is an issue for us. I don't think that the pretty-printer is rendering anything with backticks, so even if we were using them at one point, we don't seem to be now. 👍

**Comment by aryairani** [comment]
@ceedubs Oops, obvious in retrospect.  Okay, thanks!

**Pull request merged**